pkg/controller/vitessshard/reconcile_disk.go

```diff
@@ -0,0 +1,129 @@
+package vitessshard
+
+import (
+	"context"
+
+	v1 "k8s.io/api/core/v1"
+	apitypes "k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"vitess.io/vitess/go/vt/topo/topoproto"
+
+	planetscalev2 "planetscale.dev/vitess-operator/pkg/apis/planetscale/v2"
+	"planetscale.dev/vitess-operator/pkg/operator/results"
+	"planetscale.dev/vitess-operator/pkg/operator/rollout"
+)
+
+const (
+	pvcDiskSizeAnnotation = "planetscale.com/pvc-filesystem-resize"
+)
+
+func (r *ReconcileVitessShard) reconcileDisk(ctx context.Context, vts *planetscalev2.VitessShard) (reconcile.Result, error) {
+	resultBuilder := &results.Builder{}
```

Comment:
Since the whole purpose of this function is now that we maybe will set cascade, we should check here whether `vts` is already cascading, and bail out since there's no point doing all that work below. This also ensures that we don't keep sending Update attempts to set the cascade annotation when the shard is already cascading.